auacm/angular/src/app/auth.service.ts

```diff
@@ -0,0 +1,189 @@
+        resolve(new UserData());
+      });
+    });
+  };
+
+  createUser(username: string, password: string, displayName: string) : Promise<SimpleResponse> {
+    return new Promise((resolve, reject) => {
+      var params = new URLSearchParams();
+      params.append('username', username);
+      params.append('password', password);
+      params.append('display', displayName);
+
+      var headers = new Headers();
+      headers.append('Content-Type', 'application/x-www-form-urlencoded');
+
+      this._http.post('/api/create_user', params.toString(), { headers: headers }).subscribe((res: Response) => {
+        if (res.status === 200) {
+          resolve(new SimpleResponse(true, 'User created successfully.'))
+        } else {
+          resolve(new SimpleResponse(false, res.json().error));
+        }
+      }, (err: Response) => {
+        if (err.status === 401) {
+          resolve(new SimpleResponse(false, 'You need to be an admin to do this.'))
+        } else if (err.status === 400) {
+          resolve(new SimpleResponse(false, 'That user already exists.'));
+        } else {
+          resolve(new SimpleResponse(false, 'Failed to create the user.'));
+        }
+      });
+    });
+  };
+
+  changePassword(oldPassword: string, newPassword: string) : Promise<boolean> {
+    return new Promise((resolve, reject) => {
+      var params = new URLSearchParams();
+      params.append('oldPassword', oldPassword);
+      params.append('newPassword', newPassword);
+
+      var headers = new Headers();
+      headers.append('Content-Type', 'application/x-www-form-urlencoded');
+
+      this._http.post('/api/change_password', params.toString(), { headers: headers }).subscribe((res: Response) => {
+        if (res.status === 200) {
+          resolve(true);
+        } else {
+          resolve(false);
+        }
+      }, (err: Response) => {
+        resolve(false);
+      })
+    })
+  };
+
+  /**
+   * We should implement this sometime in the future so that we don't have to edit the
+   * database manually if we want to make someone an admin.
+   */
+  updateAdminStatus(username: string) : Promise<SimpleResponse> {
+    return undefined;
+  }
+
+  getRanking(timeframe: string) : Promise<RankData[]> {
+    return new Promise((resolve, reject) => {
+      var time = timeframe === undefined ? 'all' : timeframe;
+      this._http.get(`/api/ranking/${time}`).subscribe((res: Response) => {
+        var data = res.json().data;
+        var rankings = [];
+        if (res.status == 200) {
+          for (var i = 0; i < data.length; i++) {
+            rankings.push(data[i]);
+          }
+        }
+        resolve(rankings);
+      }, (err: Response) => {
+        resolve([]);
+      });
+    });
+  }
+}
+
+export class TimeFrame {
+	public DAY: string = "day";
+	public WEEK: string = "week";
+	public MONTH: string = "month";
+	public YEAR: string = "year";
+	public ALL: string = "all";
+}
+
+export class SimpleResponse {
```

Comment:
Something about this class name and location bothers me.

It seems too generic of a name to be specific to this given service. Not really a big deal, but you may want to either:
1.) Give this a more specific name that indicates it just is used in this service.
2.) Move it into a model file.